sort/patiencesort.go

```diff
@@ -0,0 +1,90 @@
+// Package sort
+// Patience sorting is a sorting algorithm inspired by the card game patience.
+//
+// For more details check out those links below here:
+// GeeksForGeeks article : https://www.geeksforgeeks.org/patience-sorting/
+// Wikipedia article: https://en.wikipedia.org/wiki/Patience_sorting
+// authors [guuzaa](https://github.com/guuzaa)
+// see patiencesort.go
+package sort
+
+import "github.com/TheAlgorithms/Go/constraints"
+
+func Patience[T constraints.Ordered](arr []T) []T {
+	if len(arr) <= 1 {
+		return arr
+	}
+
+	var piles [][]T
+
+	// Traverse the given array
+	for _, card := range arr {
+		// Iterate over all the piles
+		// and check the top of the stack of each pile
+		// is less than the current element or not.
+		left, right := 0, len(piles)
+		for left < right {
+			mid := left + (right-left)>>1
+			if piles[mid][len(piles[mid])-1] >= card {
+				right = mid
+			} else {
+				left = mid + 1
+			}
+		}
+
+		// If all the top of the stack of each pile is less than than the current element
+		if left == len(piles) {
+			// Create a new pile with the current element
+			// and insert the pile into all the piles
+			piles = append(piles, []T{card})
+		} else {
+			// The current element is placed on the leftmost existing pile 
+			// whose top card has a value greater than or equal to the new card's value
+			piles[left] = append(piles[left], card)
+		}
+	}
+
+	return mergePiles(piles)
+}
+
+func mergePiles[T constraints.Ordered](piles [][]T) []T {
+	var ret []T
+
+	// Find the smallest element of top of pile
+	// and remove it from the piles
+	// and store it into the final array , in every iteration,
+	// util all piles are empty
```

Comment:
Not necessary to explain the code. Your code is the explanation of what is happening, comment should be used only when you have to explain why you're doing something that might not be quite straight forward.

Comments should explain "why" not "what" (IMO)